src/tests/Microsoft.PowerFx.Core.Tests.Shared/IntellisenseTests/SuggestTest.cs

```diff
@@ -539,6 +539,22 @@ public void SuggestAggregateReturnType(string expression, params string[] expect
             Assert.Equal(expected, suggestion);
         }
 
+        [Theory]
+        [InlineData("Filter(TableLoop,|")]
+        [InlineData("Filter(Accounts,|")]
+        public void LazyTypesStackOverflowTest(string expression)
+        {
+            var lazyInstance = new LazyRecursiveRecordType();
+            var config = PowerFxConfig.BuildWithEnumStore(
+                new EnumStoreBuilder(),
+                new TexlFunctionSet(new[] { BuiltinFunctionsCore.Filter }));
+
+            var suggestions = SuggestStrings(expression, config, null, lazyInstance);
+
+            // Just check that the execution didn't stack overflow.
+            Assert.True(suggestions.Any());
```

Comment:
Can we merge with above?  TestSuggestLazyTypes

And test an expected result (not just Any())